### DISCUSSION TITLE
2024-06-13(목)

### DISCUSSION THREAD

**Discussion by Seoin02** [Together-3team/petFrontend#138] [데일리스크럼]
아래 내용을 복사해 작성해주세요.
```
### 완료한 일

### 해야할 일

### 어려운 일
```

**Comment by Seoin02** [comment]
### 완료한 일
- access token 쿠키에 넣기

### 해야할 일
- axois intercepter 수정&미들웨어 설정
- 로그아웃 처리
- 쿠키 옵션 설정

### 어려운 일
- 저희가 http를 써서 쿠키 보안 옵션을 거의 설정하면 안되더라고요... 괜찮을까요?
- axios intercepter을 미들웨어로 쓰려고 하는데 next 미들웨어를 따로 써야할까요?
- 이외에 인증 관련해서 해야할 일이 또 있을까요??

**Reply by devwqc** [reply]
> 저희가 http를 써서 쿠키 보안 옵션을 거의 설정하면 안되더라고요... 괜찮을까요?

제가 알기로는 `https`는 클라이언트보다 서버가 더 중요한 걸로 아는데 저희는 로컬이 `http`인 거지 실제로 사용할 서버는 아마 버셀이 될 거 같아서 `https`입니다!

AWS도 배포하긴 했지만 다들 취업하실 때 포트폴리오로 링크 제출하려면 무료로 평생 열어둘 수 있는 버셀로 하는 게 나을 거 같아요

`http`, `https` 관련 자료 첨부합니다.
https://kukim.tistory.com/152

> axios intercepter을 미들웨어로 쓰려고 하는데 next 미들웨어를 따로 써야할까요?

우선 둘이 너무 다른 성격이여서 `axiosMiddleware`를 `axiosInterceptor`로 이름 바꾸는 게 나을 거 같아요...!
`interceptor`는 말 그대로 `axios`로 request 보낼 때나 response 받을 때 사전에 어떤 작업을 하기 위함이고 next `middleware`는 SSR, SSG 보다 먼저 작동해서 인가 처리 같은 걸 할 수 있는 건데 굳이 `middleware` 따로 안 만들고 인가 처리 필요한 페이지에서 SSR이나 CSR에서 인가 처리 하면 될 거 같습니다

>이외에 인증 관련해서 해야할 일이 또 있을까요??

CSR이랑 SSR, SSG 환경에서 axios 호출 보냈을 때 쿠키에 있는 토큰 값이 잘 담겨서 요청 보내지는 지 확인하시면 될 거 같아요

**Reply by Seoin02** [reply]
헐... 감사합니다ㅠㅠ 진짜 멋진 우리 팀장님 최고!!!

**Comment by devwqc** [comment]
### 완료한 일
- `BottomSheet` 원리 분석 후 작업 진행 중

### 해야할 일
- `BottomSheet`를 포함한 모달 오늘 완료 예정, 금 ~ 일 모든 페이지 API까지 완료 예정

### 어려운 일
- 파이팅팅!! 💯💪

**Comment by q45402sk** [comment]
### 완료한 일

### 해야할 일
상품 상세 페이지

### 어려운 일

**Comment by minyoung0503** [comment]
### 완료한 일
리뷰 작성 페이지 UI

### 해야할 일
리뷰 작성 페이지 API 연결

### 어려운 일
리뷰 작성 페이지 API 연결

**Comment by seolsis** [comment]
### 완료한 일
- 버튼 클릭 시 sessionStorage에 데이터 저장, useQuery 사용 
### 해야할 일
- 옵션 api 불러오기
- header 불러오기
- 버튼 클릭 시 put

### 어려운 일
- 데이터가 한 개만 불러와지지 않아도 렌더링이 안되나요?
